spdmlib/src/message/version.rs

```diff
@@ -108,14 +106,25 @@ impl SpdmCodec for SpdmVersionResponsePayload {
             },
             MAX_SPDM_VERSION_COUNT,
         );
-        for version in versions
-            .iter_mut()
-            .take(version_number_entry_count as usize)
-        {
-            *version = SpdmVersionStruct::read(r)?;
+
+        let mut version_count = 0;
+
+        for _ in 0..version_number_entry_count {
+            if let Some(ver) = SpdmVersionStruct::read(r) {
+                if version_count < MAX_SPDM_VERSION_COUNT {
+                    versions[version_count] = ver;
+                    version_count += 1;
+                } else {
+                    // the buffer is full now, stop for scaning more versions
+                    break;
+                }
+            } else {
+                // for unknown versions, just ignore it!
```

Comment:
I recommend we only ignore the higher version - 1.4, 2.0.
The unknown lower version - 0.0, 0.9, should be rejected.

Reply:
> I recommend we only ignore the higher version - 1.4, 2.0. The unknown lower version - 0.0, 0.9, should be rejected.

updated.